lib/view_model/widgets_view_models/comments_view_model.dart

```diff
@@ -37,31 +50,47 @@ class CommentsViewModel extends BaseModel {
     await getComments();
   }
 
-  /// This methods fetch all comments on the post.
-  /// The function uses `getCommentsForPost` method by Comment Service.
-  Future getComments() async {
+  /// This function is used to get all comments on the post.
+  ///
+  /// To verify things are working, check out the native platform logs.
+  /// **params**:
+  ///   None
+  ///
+  /// **returns**:
+  /// * `Future<void>`: promise that will be fulfilled when comments are fetched.
+  ///
+  Future<void> getComments() async {
     setState(ViewState.busy);
-    final List commentsJSON =
-        await _commentService.getCommentsForPost(_postID) as List;
+    List commentsJSON = await _commentService.getCommentsForPost(_postID);
     print(commentsJSON);
+    commentsJSON = commentsJSON.reversed.toList();
     commentsJSON.forEach((commentJson) {
       _commentlist.add(Comment.fromJson(commentJson as Map<String, dynamic>));
     });
     setState(ViewState.idle);
   }
 
-  /// This function add comment on the post.
-  /// The function uses `createComments` method provided by Comment Service.
+  /// This function add comment on the post. The function uses `createComments` method provided by Comment Service.
+  ///
+  /// **params**:
+  /// * `msg`: The comment text.
   ///
-  /// params:
-  /// * `msg` : text of the comment to add.
-  Future createComment(String msg) async {
+  /// **returns**:
+  /// * `Future<void>`: promise that will be fulfilled when comment is added.
+  ///
+  Future<void> createComment(String msg) async {
     print("comment viewModel called");
     await _commentService.createComments(_postID, msg);
     addCommentLocally(msg);
   }
 
-  // This function add comment locally.
+  /// This function add comment locally.
+  ///
+  /// **params**:
+  /// * `msg`: BuildContext, contain parent info
+  ///
+  /// **returns**:
+  ///   None
   void addCommentLocally(String msg) {
     _postService.addCommentLocally(_postID);
     final creator = _userConfig.currentUser;
```

Comment:
@imshivam-gupta Is this function sending the latest added comment to server or just adding it locally. If its adding the comment locally please send it to server.

Reply:
it's sending to the server for permanent changes. Local addition is done for UI purpose to show the added comment at that moment.

Reply:
Another function was called using `_commentService` to add the latest comment to server also. I will comment the line in next comment.